amplify/backend/function/fetchDomainBalance/src/api/graphql/schemas.js

```diff
@@ -0,0 +1,174 @@
+const tokenFragment = /* GraphQL */ `
+  fragment LightTokenFragment on Token {
+    chainMetadata {
+      chainId
+      network
+    }
+    createdAt
+    decimals
+    id
+    name
+    symbol
+    type
+    updatedAt
+  }
+`;
+
+module.exports = {
+  saveTokenExchangeRate: /* GraphQL */`
+    mutation SaveTokenExchangeRate($input: CreateTokenExchangeRateInput!) {
+      createTokenExchangeRate(input: $input) {
+        tokenId
+        date
+        marketPrice {
+            currency
+            rate
+          }
+      }
+    }
+  `,
+  getTokenExchangeRate: /* GraphQL */`
+    query GetTokenExchangeRateByTokenId(
+      $tokenId: String!
+      $date: String
+      $filter: ModelTokenExchangeRateFilterInput
+    ) {
+      tokenExhangeRateByTokenId(
+        tokenId: $tokenId
+        date: { eq: $date }
+        filter: $filter
+      ) {
+        items {
+          tokenId
+          date
+          marketPrice {
+            currency
+            rate
+          }
+        }
+        nextToken
+      }
+    }
+  `,
+  getColonyDomains: /* GraphQL */`
+    query GetColonyDomains(
+      $colonyAddress: ID!
+    ) {
+      listDomains(
+        filter: {
+          colonyId: { eq: $colonyAddress }
+        }
+      ) {
+        items {
+          nativeFundingPotId
+          nativeId
+          colonyId
+          id
+          isRoot
+          # probably we need to rootId when nested teams is in place
+        }
+      }
+    }
+  `,
+  getColonyFundsClaims: /* GraphQL */`
+    ${tokenFragment}
+    query GetColonyFundsClaims(
+      $colonyAddress: ID!
+      $nextToken: String
+      $limit: Int
+    ) {
+      listColonyFundsClaims(
```

Comment:
This model needs an index on `colonyFundsClaimsId` field as well, and use the indexed query.

There are gonna be many funds claims in the DB and it'll end up making multiple requests before it selects the relevant ones.